### PR TITLE
Rename prop title to heading in Modal

### LINKED ISSUE #802: Modal title should be modal heading
Rename prop to heading. modal title should be modal heading

### PR DESCRIPTION
**Pull request by aswinshenoy**
Fixes #802

### Additional description
Renamed the prop ```title``` of Modal Component to ```heading```

---

### CONTRIBUTOR checklist (do not remove)
Please complete for every pull request

* [x] First-time contributors should sign the Contributor License Agreement. It's a fancy way of saying that you are giving away your contribution to this project. If you haven't before, wait a few minutes and a bot will comment on this pull request with instructions.
* [x] `npm run lint:fix` has been run and linting passes.
* [ ] Mocha, Jest (Storyshots), and `components/component-docs.json` CI tests pass (`npm test`).
* [ ] Tests have been added for new props to prevent regressions in the future. See [readme](https://github.com/salesforce/design-system-react/blob/master/tests/README.md).
* [x] Review the appropriate Storybook stories. Open [http://localhost:9001/](http://localhost:9001/).
* [ ] The Accessibility panel of each Storybook story has 0 violations (aXe). Open [http://localhost:9001/](http://localhost:9001/).
* [x] Review tests are passing in the browser. Open [http://localhost:8001/](http://localhost:8001/).
* [ ] Review markup conforms to [SLDS](https://www.lightningdesignsystem.com/) by looking at [DOM snapshot strings](https://facebook.github.io/jest/docs/en/snapshot-testing.html).

### REVIEWER checklist (do not remove)

* [ ] TravisCI tests pass. This includes linting, Mocha, Jest, Storyshots, and `components/component-docs.json` tests.
* [ ] Tests have been added for new props to prevent regressions in the future. See [readme](https://github.com/salesforce/design-system-react/blob/master/tests/README.md).
* [ ] Review the appropriate Storybook stories. Open [http://localhost:9001/](http://localhost:9001/).
* [ ] The Accessibility panel of each Storybook story has 0 violations (aXe). Open [http://localhost:9001/](http://localhost:9001/).
* [ ] Review tests are passing in the browser. Open [http://localhost:8001/](http://localhost:8001/).
* [ ] Review markup conforms to [SLDS](https://www.lightningdesignsystem.com/) by looking at [DOM snapshot strings](https://facebook.github.io/jest/docs/en/snapshot-testing.html).
###### Required only if there are markup / UX changes
* [ ] Add year-first date and commit SHA to `last-slds-markup-review` in `package.json` and push.
* [ ] Request a review of the deployed Heroku app by the Salesforce UX Accessibility Team.
* [ ] Add year-first review date, and commit SHA, `last-accessibility-review`, to `package.json` and push.
* [ ] While the contributor's branch is checked out, run `npm run local-update` within locally cloned [site repo](https://github.com/salesforce-ux/design-system-react-site) to confirm the site will function correctly at the next release.

### CODE REVIEW

**Comment by interactivellama** [comment]
Please make this backward compatible with `title` usage. Something like `props.title || props.heading`. You may need an ESlint disable comment.

**Comment by aswinshenoy** [comment]
I will work on it asap, and add it on `check-props` to alert about depreciation.

**Comment by aswinshenoy** [comment]
Hey @interactivellama,
 
I have added backward compatibility, and it now shows a warning too -
image

**Comment by interactivellama** [comment]
Looks like a lint issue. Please review `npm run lint` output.

File: components/modal/check-props.js [hunk 1]
```diff
@@ -18,6 +18,7 @@ if (process.env.NODE_ENV !== 'production') {
 			"assistiveText['closeButton']",
 			createDocUrl('assistiveText')
 		);
+		deprecatedProperty(COMPONENT, props.title, 'title', 'heading');
```
Comment (interactivellama) [inline components/modal/check-props.js]: It's not documented, but this should have the `createDocUrl('heading')` as the fifth parameter.

**Pull request merged**